### PR TITLE
[serve] Add test for Gradio queues (which use websockets)

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Adds a test for the Gradio queue interface which uses websockets under the hood.

Rather than reaching into the implementation details, I'm using the `gradio_client` package to make the requests, so also updated the existing tests to use this.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by shrekris-anyscale — APPROVED** [review]
Nice work!

**Review by zcin — APPROVED** [review]
Cool!

File: python/ray/serve/tests/test_websockets.py [hunk 1]
```diff
@@ -152,5 +153,51 @@ async def ws_hi(self, ws: WebSocket):
...
+            super().__init__(lambda: g.queue())
+
+    serve.run(GradioGenerator.bind())
+
+    print("Client connecting")
+    client = Client("http://localhost:8000")
+    print("Client connected")
+
+    job1 = client.submit(3, api_name="/predict")
+    print("Job2 submitted")
```
Comment (GeneDer) [inline python/ray/serve/tests/test_websockets.py:189]: Maybe this meant to be job1 🙃

**Review by GeneDer — APPROVED** [review]
LGTM

**Review by matthewdeng — APPROVED** [review]

**Comment by chris-aeviator** [comment]
@edoakes 

I think this does not test real gradio queue usage. When trying to load  a queue, gradio fials with a dns resolution error.
    ```
 # def gradio_summarizer_builder..
      return gr.Interface(
             fn=run_tasks,
             inputs=[gr.Textbox(value=example_input, label="Input prompts")],
             outputs=[gr.Textbox(label="Model output")],
         ).queue()
    
     app = GradioServer.options(
             # route_prefix="/xjd3",
             ray_actor_options={"num_gpus": 0.25}
         ).bind(
             gradio_summarizer_builder
    )
```

grafik

**Pull request merged**